### PR TITLE
fix: refactor counter logic for ordered lists

### PR DESCRIPTION
**Pull request by Vahid1919**
## Description:

Replace the use of counters() with counter(). counters() is unnecessary here since we explicitly define three variables anyway. This occasionally caused issues in Chrome when multiple lists were present in CMS test pages. Switching to counter() avoids these edge-case bugs and makes the code more readable.

## Definition of Reviewable:

- [ ] Documentation is created/updated
- [ ] Migration Guide is created/updated

- [ ] E2E tests (features, a11y, bug fixes) are created/updated

- [ ] Stories (features, a11y) are created/updated
- [ ] relevant tickets are linked

### CODE REVIEW

File: .changeset/sour-books-watch.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@solid-design-system/styles': patch
+---
+
+Replaced counters() with counter() inside ordered lists.
```
Comment (mariohamann) [inline .changeset/sour-books-watch.md]: Please provide an info why this was done.
